loihi-dl/cifar10_convnet.ipynb

```diff
@@ -0,0 +1,775 @@
+    "    loihi_net.config[nengo.Ensemble].max_rates = nengo.dists.Choice([100])\n",
+    "    loihi_net.config[nengo.Ensemble].intercepts = nengo.dists.Choice([0])\n",
+    "    loihi_net.config[nengo.Connection].synapse = 0.01  # can also try 0.005\n",
+    "\n",
+    "    inp = nengo.Node(present_images, label='input')\n",
+    "    inp_p = nengo.Probe(inp)\n",
+    "    xx = [inp]\n",
+    "    xslices = [ImageSlice(input_shape)]\n",
+    "    cores_used = 0\n",
+    "\n",
+    "    for fn_layer, neuron_layer in layers:\n",
+    "        func = fn_layer.tensor_func\n",
+    "        name = func.name\n",
+    "\n",
+    "        # --- create neuron layer\n",
+    "        yy = []\n",
+    "        yslices = []\n",
+    "        if neuron_layer is not None:\n",
+    "            layer_neurons = neuron_layer.ensemble.neuron_type\n",
+    "            if isinstance(layer_neurons, SoftLIFRate):\n",
+    "                layer_neurons = nengo.LIF(\n",
+    "                    tau_rc=layer_neurons.tau_rc,\n",
+    "                    tau_ref=layer_neurons.tau_ref,\n",
+    "                    amplitude=layer_neurons.amplitude)\n",
+    "            elif isinstance(layer_neurons, nengo.RectifiedLinear):\n",
+    "                layer_neurons = nengo.SpikingRectifiedLinear(\n",
+    "                    amplitude=layer_neurons.amplitude)\n",
+    "            else:\n",
+    "                raise ValueError(\"Unsupported neuron type %s\" % layer_neurons)\n",
+    "\n",
+    "            image_shape = neuron_layer.image_shape\n",
+    "            if not neuron_layer.on_chip:\n",
+    "                y = nengo.Ensemble(\n",
+    "                    image_shape.size, 1,\n",
+    "                    neuron_type=layer_neurons,\n",
+    "                    label=\"%s\" % (func.name,))\n",
+    "                loihi_net.config[y].on_chip = False\n",
+    "                yy.append(y.neurons)\n",
+    "                yslices.append(ImageSlice(image_shape))\n",
+    "            else:\n",
+    "                split_slices = image_shape.split_channels(\n",
+    "                    max_size=1024, max_channels=8)\n",
+    "                for image_slice in split_slices:\n",
+    "                    assert image_slice.size <= 1024\n",
+    "                    idxs = image_slice.channel_idxs()\n",
+    "                    y = nengo.Ensemble(\n",
+    "                        image_slice.size, 1,\n",
+    "                        neuron_type=layer_neurons,\n",
+    "                        label=\"%s_%d:%d\" % (func.name, min(idxs), max(idxs)))\n",
+    "                    yy.append(y.neurons)\n",
+    "                    yslices.append(image_slice)\n",
+    "                    cores_used += 1\n",
+    "        else:\n",
+    "            output_shape = func.output_shape(func.shape_in)\n",
+    "            assert output_shape.size == fn_layer.size_out\n",
+    "            if 0:  # node works on emulator, on Loihi we need something on-chip\n",
```

Comment:
Would be nice to avoid the `if 0`. I am guessing we do this because it's faster to do this snippet on the emulator, but the `else` works on the emulator too, so can we just remove this block? Alternatively, I think we should set the `target` explicitly in the notebook.